File: app/src/main/java/com/app/l_pesa/loanplan/model/ResLoanPlans.java
package com.app.l_pesa.loanplan.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class ResLoanPlans {

    @SerializedName("status")
    @Expose
    private Status status;
    @SerializedName("data")
    @Expose
    private ArrayList<Datum> data = null;

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public ArrayList<Datum> getData() {
        return data;
    }

    public void setData(ArrayList<Datum> data) {
        this.data = data;
    }

    public class Item {

        @SerializedName("details")
        @Expose
        private Details details;

        public Details getDetails() {
            return details;
        }

        public void setDetails(Details details) {
            this.details = details;
        }


    }


    public class Status {

        @SerializedName("statusCode")
        @Expose
        private Integer statusCode;
        @SerializedName("isSuccess")
        @Expose
        private Boolean isSuccess;
        @SerializedName("message")
        @Expose
        private String message;

        public Integer getStatusCode() {
            return statusCode;
        }

        public void setStatusCode(Integer statusCode) {
            this.statusCode = statusCode;
        }

        public Boolean getIsSuccess() {
            return isSuccess;
        }

        public void setIsSuccess(Boolean isSuccess) {
            this.isSuccess = isSuccess;
        }

        public String getMessage() {
            return message;
        }

        public void setMessage(String message) {
            this.message = message;
        }

    }

    public class Details {

        @SerializedName("product_id")
        @Expose
        private Integer productId;
        @SerializedName("loan_amount")
        @Expose
        private Integer loanAmount;
        @SerializedName("required_credit_score")
        @Expose
        private Integer requiredCreditScore;
        @SerializedName("loan_period")
        @Expose
        private Integer loanPeriod;
        @SerializedName("loan_period_type")
        @Expose
        private String loanPeriodType;
        @SerializedName("loan_interest_rate")
        @Expose
        private Double loanInterestRate;
        @SerializedName("status")
        @Expose
        private String status;
        @SerializedName("btnStatus")
        @Expose
        private String btnStatus;
        @SerializedName("btnColor")
        @Expose
        private String btnColor;
        @SerializedName("btnHexColor")
        @Expose
        private String btnHexColor;
        @SerializedName("btnText")
        @Expose
        private String btnText;
        @SerializedName("alertMgs")
        @Expose
        private String alertMgs;
        @SerializedName("wizardStatus")
        @Expose
        private String wizardStatus;
        @SerializedName("wizardURL")
        @Expose
        private String wizardURL;

        public Integer getProductId() {
            return productId;
        }

        public void setProductId(Integer productId) {
            this.productId = productId;
        }

        public Integer getLoanAmount() {
            return loanAmount;
        }

        public void setLoanAmount(Integer loanAmount) {
            this.loanAmount = loanAmount;
        }

        public Integer getRequiredCreditScore() {
            return requiredCreditScore;
        }

        public void setRequiredCreditScore(Integer requiredCreditScore) {
            this.requiredCreditScore = requiredCreditScore;
        }

        public Integer getLoanPeriod() {
            return loanPeriod;
        }

        public void setLoanPeriod(Integer loanPeriod) {
            this.loanPeriod = loanPeriod;
        }

        public String getLoanPeriodType() {
            return loanPeriodType;
        }

        public void setLoanPeriodType(String loanPeriodType) {
            this.loanPeriodType = loanPeriodType;
        }

        public Double getLoanInterestRate() {
            return loanInterestRate;
        }

        public void setLoanInterestRate(Double loanInterestRate) {
            this.loanInterestRate = loanInterestRate;
        }

        public String getStatus() {
            return status;
        }

        public void setStatus(String status) {
            this.status = status;
        }

        public String getBtnStatus() {
            return btnStatus;
        }

        public void setBtnStatus(String btnStatus) {
            this.btnStatus = btnStatus;
        }

        public String getBtnColor() {
            return btnColor;
        }

        public void setBtnColor(String btnColor) {
            this.btnColor = btnColor;
        }

        public String getBtnHexColor() {
            return btnHexColor;
        }

        public void setBtnHexColor(String btnHexColor) {
            this.btnHexColor = btnHexColor;
        }

        public String getBtnText() {
            return btnText;
        }

        public void setBtnText(String btnText) {
            this.btnText = btnText;
        }

        public String getAlertMgs() {
            return alertMgs;
        }

        public void setAlertMgs(String alertMgs) {
            this.alertMgs = alertMgs;
        }

        public String getWizardStatus() {
            return wizardStatus;
        }

        public void setWizardStatus(String wizardStatus) {
            this.wizardStatus = wizardStatus;
        }

        public String getWizardURL() {
            return wizardURL;
        }

        public void setWizardURL(String wizardURL) {
            this.wizardURL = wizardURL;
        }

    }

    public class Datum {

        @SerializedName("appliedProduct")
        @Expose
        private AppliedProduct appliedProduct;
        @SerializedName("item")
        @Expose
        private ArrayList<Item> item = null;

        public AppliedProduct getAppliedProduct() {
            return appliedProduct;
        }

        public void setAppliedProduct(AppliedProduct appliedProduct) {
            this.appliedProduct = appliedProduct;
        }

        public ArrayList<Item> getItem() {
            return item;
        }

        public void setItem(ArrayList<Item> item) {
            this.item = item;
        }

    }

    public class AppliedProduct {

        @SerializedName("identity_number")
        @Expose
        private String identityNumber;
        @SerializedName("loan_purpose_message")
        @Expose
        private String loanPurposeMessage;
        @SerializedName("applied_date")
        @Expose
        private String appliedDate;
        @SerializedName("loan_status")
        @Expose
        private String loanStatus;
        @SerializedName("product_id")
        @Expose
        private Integer productId;

        public String getIdentityNumber() {
            return identityNumber;
        }

        public void setIdentityNumber(String identityNumber) {
            this.identityNumber = identityNumber;
        }

        public String getLoanPurposeMessage() {
            return loanPurposeMessage;
        }

        public void setLoanPurposeMessage(String loanPurposeMessage) {
            this.loanPurposeMessage = loanPurposeMessage;
        }

        public String getAppliedDate() {
            return appliedDate;
        }

        public void setAppliedDate(String appliedDate) {
            this.appliedDate = appliedDate;
        }

        public String getLoanStatus() {
            return loanStatus;
        }

        public void setLoanStatus(String loanStatus) {
            this.loanStatus = loanStatus;
        }

        public Integer getProductId() {
            return productId;
        }

        public void setProductId(Integer productId) {
            this.productId = productId;
        }

    }
}

File: app/src/main/java/com/app/l_pesa/loanplan/model/ResLoanPlans.kt
package com.app.l_pesa.loanplan.model

import androidx.annotation.Nullable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

class ResLoanPlans {

    @SerializedName("status")
    @Expose
    var status: Status? = null
    @SerializedName("data")
    @Expose
    var data: Data? = null

    inner class Item {

        @SerializedName("details")
        @Expose
        var details: Details? = null

    }

    inner class Status {

        @SerializedName("statusCode")
        @Expose
        var statusCode: Int = 0
        @SerializedName("isSuccess")
        @Expose
        var isSuccess: Boolean = false
        @SerializedName("message")
        @Expose
        var message: String = ""

    }

    inner class AppliedProduct {

        @SerializedName("loan_id")
        @Expose
        var loanId: Int = 0
        @SerializedName("identity_number")
        @Expose
        var identityNumber: String = ""
        @SerializedName("currency_code")
        @Expose
        var currencyCode: String = ""
        @SerializedName("product_id")
        @Expose
        var productId: Int = 0
        @SerializedName("loan_amount")
        @Expose
        var loanAmount: Double = 0.0
        @SerializedName("interest_rt")
        @Expose
        var interestRt: Double = 0.0
        @SerializedName("grace_period")
        @Expose
        var gracePeriod: String = ""
        @SerializedName("interest_rate")
        @Expose
        var interestRate: String = ""
        @SerializedName("no_of_weeks")
        @Expose
        var noOfWeeks: Int = 0
        @SerializedName("duration")
        @Expose
        var duration: String = ""
        @SerializedName("convertion_dollar_value")
        @Expose
        var convertionDollarValue: Double = 0.0
        @SerializedName("convertion_loan_amount")
        @Expose
        var convertionLoanAmount: Int = 0
        @SerializedName("actual_loan_amount")
        @Expose
        var actualLoanAmount: Double = 0.0
        @SerializedName("conversion_charge")
        @Expose
        var conversionCharge: String = ""
        @SerializedName("conversion_charge_amount")
        @Expose
        var conversionChargeAmount: String = ""
        @SerializedName("processing_fees")
        @Expose
        var processingFees: String = ""
        @SerializedName("processing_fees_amount")
        @Expose
        var processingFeesAmount: String = ""
        @SerializedName("loan_purpose_message")
        @Expose
        var loanPurposeMessage: String = ""
        @SerializedName("cr_sc_when_requesting_loan")
        @Expose
        var crScWhenRequestingLoan: String = ""
        @SerializedName("disapprove_reason")
        @Expose
        var disapproveReason: String = ""
        @SerializedName("loan_status")
        @Expose
        var loanStatus: String = ""
        @SerializedName("applied_date")
        @Expose
        var appliedDate: String = ""
        @SerializedName("finished_date")
        @Expose
        var finishedDate: String = ""
        @SerializedName("disapprove_date")
        @Expose
        var disapproveDate: String = ""
        @SerializedName("lost_date")
        @Expose
        var lostDate: String = ""
        @SerializedName("sanctioned_date")
        @Expose
        var sanctionedDate: String = ""
        @SerializedName("due_date")
        @Expose
        var dueDate: String = ""

    }

    inner class Data {

        @SerializedName("appliedProduct")
        @Expose
        var appliedProduct: AppliedProduct? = null
        @SerializedName("item")
        @Expose
        var item: ArrayList<Item>? = null

    }

    inner class Details {

        @SerializedName("product_id")
        @Expose
        var productId: Int = 0
        @SerializedName("currency_code")
        @Expose
        var currencyCode: String = ""
        @SerializedName("required_credit_score")
        @Expose
        var requiredCreditScore: Int = 0
        @SerializedName("convertion_dollar_value")
        @Expose
        var convertionDollarValue: Double = 0.0
        @SerializedName("convertion_loan_amount")
        @Expose
        var convertionLoanAmount: Double = 0.0
        @SerializedName("conversion_charge")
        @Expose
        var conversionCharge: Double = 0.0
        @SerializedName("conversion_charge_amount")
        @Expose
        var conversionChargeAmount: Double = 0.0
        @SerializedName("processing_fees")
        @Expose
        var processingFees: Double = 0.0
        @SerializedName("processing_fees_amount")
        @Expose
        var processingFeesAmount: Double = 0.0
        @SerializedName("loan_amount")
        @Expose
        var loanAmount: Double = 0.0
        @SerializedName("loan_period")
        @Expose
        var loanPeriod: Int = 0
        @SerializedName("loan_period_type")
        @Expose
        var loanPeriodType: String = ""
        @SerializedName("loan_interest_rate")
        @Expose
        var loanInterestRate: Double = 0.0
        @SerializedName("status")
        @Expose
        var status: String = ""
        @SerializedName("btnStatus")
        @Expose
        var btnStatus: String = ""
        @SerializedName("btnColor")
        @Expose
        var btnColor: String = ""
        @SerializedName("btnHexColor")
        @Expose
        var btnHexColor: String = ""
        @SerializedName("btnText")
        @Expose
        var btnText: String = ""
        @SerializedName("alertMgs")
        @Expose
        var alertMgs: String = ""
        @SerializedName("wizardStatus")
        @Expose
        var wizardStatus: Boolean = false
        @SerializedName("wizardURL")
        @Expose
        var wizardURL: String = ""

        @SerializedName("actual_loan_amount")
        @Expose
        var actualLoanAmount: Double = 0.0


        @SerializedName("bannerObject")
        @Expose
        var bannerObject: BannerObject? = null

    }

    data class BannerObject(
            @SerializedName("text")
            @Expose
            var text: String = "",

            @SerializedName("fontSize")
            @Expose
            var fontSize: Int = 12,

            @SerializedName("isShimmer")
            @Expose
            var isShimmer: Boolean = true,

            @SerializedName("fontColor")
            @Expose
            var fontColor: String = "#4d0000",

            @SerializedName("backColor")
            @Expose
            var backColor: String = "#f28400",

            @SerializedName("isBanner")
            @Expose
            var isBanner: Boolean = false,


            @SerializedName("style")
            @Expose
            var style: BannerStyle? = null

    )

    data class BannerStyle(
            @SerializedName("isBanner")
            @Expose
            var type: Int = 0,

            @SerializedName("position")
            @Expose
            var position: String = "right"
    )

}